src/windows/loadsave.c

```diff
@@ -602,6 +602,7 @@ static void window_loadsave_populate_list(int includeNewItem, bool browsable, co
 	strncpy(_directory, directory, sizeof(_directory));
 	if (_extension != extension) {
 		strncpy(_extension, extension, sizeof(_extension));
+		_extension[sizeof(_extension) - 1] = '\0';
```

Comment:
I keep suggesting you make a helper function which does a strncpy + null terminator for you.